src/runtime_src/core/pcie/emulation/hw_em/alveo_shim/shim.cxx

```diff
@@ -746,6 +748,20 @@ namespace xclhwemhal2 {
         }
         setenv("VITIS_KERNEL_PROFILE_FILENAME", kernelProfileFileName.c_str(), true);
         setenv("VITIS_KERNEL_TRACE_FILENAME", kernelTraceFileName.c_str(), true);
+
+        const char* ldisplay = std::getenv("DISPLAY");
+        if(!ldisplay)
+        {
+          if (mLogStream.is_open())
+          {
+            mLogStream << __func__ << " DISPLAY environment is not available so expect an exit from the application " << std::endl;
+            //DEBUG_MSGS_COUT(" DISPLAY environment is not available so expect an exit from the application ");
+          }
+          std::string dMsg = "ERROR: [HW-EMU 26] DISPLAY environment is not available so expect an exit from the application";
+          logMessage(dMsg, 0);
+          exit(EXIT_FAILURE);
```

Comment:
You can't just exit the process.   Throw an exception and let upper level code do what it needs to do.

Reply:
currently XRT is exiting when required environment variables are not set such as
USER, XILINX_VITIS. If xrt.ini file contains debug_mode=gui and X Display manager is not present then expecting an exit is proper, isn't it?
std::throw expection can serve the purpose as well.